### PR TITLE
[IMP] selection: alleviate size of revision on 'move header'

### PR DESCRIPTION
**Pull request by fdamhaut**
Aleviate the size  of the message dispatched on the server by grouping the dispatch per size and only if it's different from the current size.

Task: 4378896

## Description:

description of this task, what is implemented and why it is implemented that way.

Task: [4378896](https://www.odoo.com/odoo/2328/tasks/4378896)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/ui_stateful/selection.ts [hunk 1]
```diff
@@ -580,15 +580,29 @@ export class GridSelectionPlugin extends UIPlugin {
 
     const toRemove = isBasedBefore ? cmd.elements.map((el) => el + thickness) : cmd.elements;
     let currentIndex = cmd.base;
+
+    const resizing_groups: Record<number, number[]> = {};
+
     for (const element of toRemove) {
       const size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, element);
+      const current_size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
```
Comment (LucasLefevre) [inline src/plugins/ui_stateful/selection.ts:588]:
```suggestion
      const currentSize = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
```

File: src/plugins/ui_stateful/selection.ts [hunk 2]
```diff
@@ -580,15 +580,29 @@ export class GridSelectionPlugin extends UIPlugin {
 
     const toRemove = isBasedBefore ? cmd.elements.map((el) => el + thickness) : cmd.elements;
     let currentIndex = cmd.base;
+
+    const resizing_groups: Record<number, number[]> = {};
```
Comment (LucasLefevre) [inline src/plugins/ui_stateful/selection.ts:584]:
```suggestion
    const resizingGroups: Record<number, number[]> = {};
```

File: src/plugins/ui_stateful/selection.ts [hunk 3]
```diff
@@ -580,15 +580,29 @@ export class GridSelectionPlugin extends UIPlugin {
...
+
     for (const element of toRemove) {
       const size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, element);
+      const current_size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
+      if (size != current_size) {
+        if (size in resizing_groups) {
+          resizing_groups[size].push(currentIndex);
+        } else {
+          resizing_groups[size] = [currentIndex];
+        }
```
Comment (LucasLefevre) [inline src/plugins/ui_stateful/selection.ts:594]: A slightly shorter way of writing the same thing
```suggestion
        resizing_groups[size] ??= [];
        resizing_groups[size].push(currentIndex);
```

File: src/plugins/ui_stateful/selection.ts [hunk 4]
```diff
@@ -580,15 +580,26 @@ export class GridSelectionPlugin extends UIPlugin {
...
       const size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, element);
+      const currentSize = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
+      if (size != currentSize) {
+        resizingGroups[size] ??= [];
+        resizingGroups[size].push(currentIndex);
+        currentIndex += 1;
+      }
+    }
+
+    for (const size in resizingGroups) {
```
Comment (pro-odoo) [inline src/plugins/ui_stateful/selection.ts:596]: Maybe we could add a test with `spyDispatch` to ensure the elements are grouped ?

**Review by pro-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**